### ISSUE TITLE
mp4box APPCRASH when working with very large files

### ISSUE THREAD

**Issue by holozip** [gpac/gpac#174]
When working with extremely large files (5hrs+, 3.5mbit, 7.4gb), mp4box crashes with the following report:

```
Problem signature:
  Problem Event Name:    APPCRASH
  Application Name:    mp4box.exe
  Application Version:    0.0.0.0
  Application Timestamp:    54943b1e
  Fault Module Name:    StackHash_deed
  Fault Module Version:    6.1.7601.18247
  Fault Module Timestamp:    521eaf24
  Exception Code:    c0000374
  Exception Offset:    00000000000c4102
  OS Version:    6.1.7601.2.1.0.272.7
  Locale ID:    1033
  Additional Information 1:    deed
  Additional Information 2:    deed99506a703e22557440366600e92b
  Additional Information 3:    b05e
  Additional Information 4:    b05e9bf64cd26cbc9cf96d90a59878b7
```

Example command line:
`"C:\Program Files\GPAC\mp4box.exe" -v -add 20211835_3500000.ismv out.mp4`

Version info:
```
MP4Box - GPAC version 0.5.1-DEV-rev5568
GPAC Copyright (c) Telecom ParisTech 2000-2012
GPAC Configuration: (static configuration file)
Features: GPAC_64_BITS GPAC_HAS_SSL GPAC_HAS_SPIDERMONKEY GPAC_HAS_JPEG GPAC_HAS_PNG
```

At the point of crash, the following verbose messages are available:
```
[iso file] Current top box start before parsing 1578546
[iso file] Read Box type mdat size 842633 start 4296545842
[iso file] Current top box start before parsing 2421179
[iso file] Read Box type moof size 676 start 4297388475
[iso file] Read Box type mfhd size 16 start 4297388483
[iso file] Read Box type traf size 652 start 4297388499
[iso file] Read Box type tfhd size 20 start 4297388507
[iso file] Read Box type trun size 624 start 4297388527
```

It crashes in the same place every time.

c0000374 appears to be a heap overrun or heap corruption:
http://blogs.msdn.com/b/jiangyue/archive/2010/03/16/windows-heap-overrun-monitoring.aspx

Could mp4box overrunning an int boundary? The point at which it dies (4297388527) is suspiciously close to a four byte unsigned int: 4,294,967,295

**Comment by rbouqueau** [comment]
Indeed. Do you think you can share a sample file with us?

There must be something specific with this file because it's been a long time I haven't seen an issue with large files :)

**Comment by holozip** [comment]
Sure. Can I send you the url to download privately in some way?

The content itself belongs to a customer (they're happy for me to send it, but they wouldn't be happy at me making it available to the public).

**Comment by rbouqueau** [comment]
Yes. Please drop me an email.

**Comment by holozip** [comment]
Sent to your gpac-licensing.com address.

**Comment by rbouqueau** [comment]
I got the file thanks. I cannot reproduce in any way (official build, my debug build, 32 and 64 bits).

Several leads:
1) try to get the latest version from http://gpac.io/downloads/gpac-nightly-builds/, does it work?
2) check your tmp dir is not on a separate logical disk, or use the '-tmp' option
3) check none of your logical partition is in FAT32 which doesn't support large files

Is there anything we should know from your environment?

**Comment by holozip** [comment]
Thank you Romain. I had no problems using the latest nightly build:
```
d:\GCTMP\ism>..\GPAC\mp4box.exe -add V43000_3500000.ismv out.mp4
IsoMedia import V43000_3500000.ismv - track ID 1 - Video (size 1280 x 720)
Saving to out.mp4: 0.500 secs Interleaving
```

So this is perhaps something specific with version 0.5.1-DEV-rev5568. In case it helps, I've zipped up a copy of the distribution that fails and placed it here:
http://54.220.152.216/GPAC.zip

For questions 2 and 3; I made sure we used a tmp dir on the same logical disk, the same problem occurred. All volumes are NTFS.

I'll update our servers to use the new build.

Thank you very much for your help!
